### DISCUSSION TITLE
使用 Vercel 托管建立自己的简悦外部链接资料库 · 建站方案

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3960] [服务]
![External v2](https://user-images.githubusercontent.com/81074/169682369-7ac181d8-5096-4cac-8899-e13d70a311a6.png)

> 来自 [简悦社区](https://t.me/simpread) 用户 [OverflowCat](https://github.com/OverflowCat) Vercel 版本。

# 描述

因为简悦基于 Local First 构建的存储体系，其导出文件全部都会存储在本地同步目录的 output 目录下。如果我们开启了导出 HTML 功能，则阅读模式下的文章就会离线备份到本地。

由于这样的机制，我们可以尝试把 output 路径下 HTML 文件托管到网络上，使用成熟的静态文件托管服务把 Local 的文档备份变成在线的资料库，而令人可喜的是，这样的静态托管服务很多，而且大多数都有免费的套餐，个人使用基本不会产生费用。

结合简悦的 [自动化](http://ksria.com/simpread/docs/#/自动化) 和 [外部链接](https://github.com/Kenshin/simpread/discussions/3725#discussioncomment-2691470) 功能，我们也很容易在阅读文章的过程中，随时把标注的内容输出与输出的外部链接绑定其实，实现从阅读笔记到外部资料库的深度跳转。

# 源代码

https://github.com/OverflowCat/simpublish

# 教程

https://github.com/Kenshin/simpread/discussions/3960#discussioncomment-2795341

# 演示

- https://simpublish-demo.vercel.app 密码：`123456`

# 视频

https://user-images.githubusercontent.com/81074/169693835-839a680c-192a-4bb2-bd8d-1481a8423318.mp4

# 功能一览

- [x] 密码验证 
- [x] 阅读列表 
- [x] 正文（可直接读取 `idx` 或 `title`） 
- [x] 自定义样式 
- [x] 暗色模式 
- [ ] 更多自定义

# 区别

[官方资料库（阅读列表）](https://kb.simpread.pro/#/page/阅读列表) · [静态部署](https://github.com/Kenshin/simpread/discussions/3823) · [建站方案](https://github.com/Kenshin/simpread/discussions/3960) 的区别 👇 

|        | 阅读列表 | 文章页 | 密码 | API | 隐私                 | 环境准备                | 难易程度          | 备注             |
|--------|------|-----|----|-----|--------------------|---------------------|---------------|----------------|
| 官方方案   | ✔    | ✔   | ✔  | ✔   | 数据在用户的坚果云中，简悦只负责读取 | 不需要                 | 需要授权以及开通权限  [教程](https://kb.simpread.pro/#/page/开通开放平台) | 不需要任何额外环境      |
| 静态部署方案 | ✔    | ✖   | ✖  | ✖   | 数据在用户的 Github      | Git、Github、Vercel 等 | 简单            | 需要掌握 git 等操作环境 |
| 建站方案   | ✔    | ✔   | ✔  | ✖   | 数据在用户的 Github      | Git、Github、Vercel 等 | 中等            | 需要掌握 git 等操作环境 |

# 进阶教程

https://github.com/Kenshin/simpread/discussions/3960#discussioncomment-2798218

# 附录

1. [静态部署方案](https://github.com/Kenshin/simpread/discussions/3823)

2. [官方资料库方案](https://kb.simpread.pro/#/page/建立资料库)

# 关联

- [使用静态网页托管建立自己的简悦外部链接资料库](https://github.com/Kenshin/simpread/discussions/3823)

**Comment by Kenshin** [comment]
# 教程

按照下面的教程，即便是没有任何编程基础的用户也能使用。

## 准备

1. Github 账户

2. Git 环境

3. Vercel 账户

## 设置

设置有两种方式：

1. [Fork 方案](https://github.com/Kenshin/simpread/discussions/3960#discussioncomment-2797923) **首选方案**

   - 优点：当主程序更新后，可手动自动更新。
   - 缺点：相对复杂

2. [Deploy 方案](https://github.com/Kenshin/simpread/discussions/3960#discussioncomment-2797922)

   - 优点：简单
   - 缺点：当主程序更新后，无法自动化更新主程序

## 发布

进入 `https://vercel.com/<你的 Vercel 账户>/<起的名字>/settings/git` ，打开在 Github 上面的 repo

![image](https://user-images.githubusercontent.com/81074/169643170-b8c05dc0-641e-4ad6-b8aa-3ff8b3dba6c4.png)

点击 **Code** 复制 repo 地址

![image](https://user-images.githubusercontent.com/81074/169643357-d4a44efd-5cbb-4794-b847-5038e4ce5902.png)

在本地使用 `git clone git@github.com:<username>/<title>.git` Clone 到本地。

手动建立  `_output` （如果没有的话）

![image](https://user-images.githubusercontent.com/81074/169643330-12e9d8bc-5900-477b-80da-5a692aaab4eb.png)

至此，全部的环境已经准备完毕，当   `_output` 出现 HTML 文件后，使用 `git` 提交后，进入 https://simpublish-demo.vercel.app 就能看到你刚才发布的内容了。

![image](https://user-images.githubusercontent.com/81074/169643444-e6b02439-8a42-4f82-8517-bba2f51a399a.png)

## 导出全文

### 手动方案

1. 通过简悦 **手动导出 HTML 到本地**  然后手动转移到 `_output`

    ![image](https://user-images.githubusercontent.com/81074/169684544-1f986efa-4c15-42a9-b0c6-3b88cb46d136.png)

2. 通过简悦 [自动化](http://ksria.com/simpread/docs/#/自动化) 服务，当 **加入稍后读自动导出 HTML**  然后手动转移到 `_output`

   ![image](https://user-images.githubusercontent.com/81074/169684619-690f9cf3-9be4-4720-9d99-32dff9b58beb.png)

### 自动化方案

可以利用 Hazel 或 Windows 的 [DropIt](http://www.dropitproject.com/) 实现自动化导出到 `_output`。

下图是利用 Hazel 的一个方案。

![image](https://user-images.githubusercontent.com/81074/169693098-11005dc4-eabd-4844-a5bf-d1d878d3f66a.png)

```
cd /Users/xxx/simpublish-demo
git add .
git commit -m "auto save:Update some html pages."
git push origin main
```

## 导出标注

在任意阅读页面进行标注操作，然后自动导入到 Obsidian 并将外部链接指向刚刚设置的资料库。

https://github.com/Kenshin/simpread/discussions/3960#discussioncomment-2798122

## 升级

1. [Deploy 方案](https://github.com/Kenshin/simpread/discussions/3960#discussioncomment-2797922) 页面最底部

2. [Fork 方案](https://github.com/Kenshin/simpread/discussions/3960#discussioncomment-2797923) 页面最底部

**Comment by Kenshin** [comment]
# Deploy 方案

进入 https://github.com/OverflowCat/simpublish 并选择 **Deploy**

![image](https://user-images.githubusercontent.com/81074/169689932-1fc7583d-ef89-433c-ad54-8fd8a2c593c4.png)

稍等一会会自动进入下面界面的页面

![image](https://user-images.githubusercontent.com/81074/169642866-2a2a9ad0-436e-4361-a85e-061ade9e1235.png)

登录你的 Github 账户

![image](https://user-images.githubusercontent.com/81074/169679887-16c29249-2f55-40e7-81b2-d764ac8a0d13.png)

稍等片刻后会让你选择 **账户名称**（一般默认即可），并给这个 Repo 起名，比如我的名字叫做 `simpublish-demo`，然后选择 **Create**，稍等片刻后，会进入下图的界面。

![image](https://user-images.githubusercontent.com/81074/169642960-066cdabb-be6c-4b85-a67f-db62cd68d635.png)

设置密码（因为自建资料库一般都是很个人的行为，所以默认是需要密码才能访问的），设置后选择 **Deploy**

![image](https://user-images.githubusercontent.com/81074/169642976-e5877ba9-e45d-4116-acfd-a29838ff4afb.png)

稍等一会会进入下面的界面

![image](https://user-images.githubusercontent.com/81074/169643006-2a9016f9-617e-4991-8e82-c18b743a1df6.png)

然后就开始进入自动构建环节

https://user-images.githubusercontent.com/81074/169643068-ea50698a-bd6d-4ca7-a2d1-e80743c62857.mp4

如果之前设置都正确的话，会看到下面的界面

![image](https://user-images.githubusercontent.com/81074/169643075-bb25cf77-bc47-4648-b31a-0a38ffb5b4d3.png)

点击进入 **Dashboard**，进入下图的界面并访问 **Visit**

![image](https://user-images.githubusercontent.com/81074/169648096-83de07f6-2770-4d03-8401-4f24a1c8cec5.png)

进入 `https://<你之前设置的名称>.vercel.app` 比如我的就是 https://simpublish-demo.vercel.app 输入之前设定的密码

![image](https://user-images.githubusercontent.com/81074/169643109-199416b6-8c45-43df-a821-8df7135cca8c.png)

最终会看到下图的界面

![image](https://user-images.githubusercontent.com/81074/169679921-ada77398-acf7-4529-b6c5-a9e84fd1054d.png)

## 升级

使用 https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/merging-an-upstream-repository-into-your-fork 手动方式（需要有一定的 git 操作能力）

![image](https://user-images.githubusercontent.com/81074/169692003-dc0dc59f-aa99-401e-b480-89e5f92ef846.png)

```
cd simpread-demo
git pull https://github.com/OverflowCat/simpublish main
```

**Comment by Kenshin** [comment]
# Fork 方案

进入 https://github.com/OverflowCat/simpublish 选择 **Fork**

![image](https://user-images.githubusercontent.com/81074/169679968-d8125710-febf-47c0-96d7-ae17d3962b54.png)

起个名字叫 `simpublish-demo-fork` 后点击 **Create fork**

![image](https://user-images.githubusercontent.com/81074/169679987-7d793574-4eb8-4dd2-adda-b1eb0e554ade.png)

稍等片刻会自动进入 `https://github.com/<你的 Gihbub 账户>/simpublish-demo-fork`

![image](https://user-images.githubusercontent.com/81074/169680041-511ef114-09aa-4224-9c2a-7198dc2d757f.png)

进入 https://vercel.com/dashboard  选择 **New Project**

![image](https://user-images.githubusercontent.com/81074/169680076-9cfa11e0-a3b9-4462-8e3c-778ac9ca89df.png)

然后选择 `simpublish-demo-fork` 这个 Repo

![image](https://user-images.githubusercontent.com/81074/169680185-05c2fbd8-8ee3-4e53-8793-2fefb3464831.png)

稍等会进入下图的界面

![image](https://user-images.githubusercontent.com/81074/169680240-b5a805db-541f-4ad2-b963-1fe87086055e.png)

除密码外其余都不需要设置，密码的 `NAME` 设置为 `SIMPUBLISH_PASSWD` 然后选择 **Add**

![image](https://user-images.githubusercontent.com/81074/169680345-36ed7a6a-ec6b-4dff-aa09-cab15b373eb6.png)

设置完毕后选择 **Deploy**，然后就开始进入自动构建环节

https://user-images.githubusercontent.com/81074/169643068-ea50698a-bd6d-4ca7-a2d1-e80743c62857.mp4

如上述无误后，会提示 **Congratulations** 并进入下图界面

![image](https://user-images.githubusercontent.com/81074/169680382-76c1c254-7674-44bf-a627-63996825cd81.png)

点击 **Go to Dashboard** 进入下图页面后选择 **Visit**

![image](https://user-images.githubusercontent.com/81074/169680426-a12d0fac-ed07-4fa7-8e1c-20a4e6f39002.png)

进入 `https://<你之前设置的名称>.vercel.app` 比如我的就是 https://simpublish-demo.vercel.app 输入之前设定的密码

![image](https://user-images.githubusercontent.com/81074/169680429-5f6e659d-0e93-4cd7-a34b-7005af007d08.png)

最终会看到下图的界面

![image](https://user-images.githubusercontent.com/81074/169680441-523c14ed-aedd-4bd1-b686-6bc08a622326.png)

## 升级

当 https://github.com/OverflowCat/simpublish 更新后，可以通过 **Fetch upstream** 自动升级

![image](https://user-images.githubusercontent.com/81074/169684406-4c0e1bb7-d268-46f8-bd1f-279cb0f1abeb.png)

**Comment by Kenshin** [comment]
# 使用

简悦中有一个功能叫做【外部链接】，为了让简悦的【外部链接】指向刚刚建立的资料库，需要做一些设置：

1. 首先需要安装/升级 [Markdown 辅助增加插件](https://simpread.ksria.cn/plugins/details/HD9GmoatXd) 到最新版。

2. 然后配置【简悦】→【服务】→【外部链接】为 `https://<site_name>.vercel.app/articles/{{id}}` （ 根据上面的例子，这个链接应该是 https://simpublish-demo.vercel.app/articles/2073

3. 在【Markdown 辅助增加插件】的模板中可以使用 `{{an_ext_uri}}` 占位符作为外部链接。

将上面的内容复制到选项页 → 服务 → 定制导出

![image](https://user-images.githubusercontent.com/81074/169684502-160caa7f-6b63-44d6-b150-2a61d8671934.png)

然后在 Markdown 辅助增强插件中就会实现下图的功能。

![image](https://user-images.githubusercontent.com/81074/169684070-3e71754e-7abd-4ab0-ba8d-3f12d31f29be.png)

## 进阶

如果你是 Obsidian 用户，可以看此 [一站式教程](https://github.com/Kenshin/simpread/discussions/3932)。

**Reply by SEAHIVE** [reply]
不好意思，请问下离线页面怎么才会存到GitHub？按照fork的步骤操作了，把文章加入到稍后读并没有出现在项目的output文件夹内呀0-0

**Reply by Kenshin** [reply]
这是知识库的功能，你可以看 [一站式教程](https://www.yuque.com/kenshin/simpread/wkswh7)，但我不建议使用离线 HTML，push 时会比较慢。

仍有问题的话，请通过 https://github.com/Kenshin/simpread/issues/new/choose 新建 Issues 讨论。

**Reply by Kenshin** [reply]
@sche11 

嗨，如果想通过简悦 → md → Github 的话，没必要这么折腾，简悦本就支持将阅读模式的内容以 md 形式导入到 Github，细节 [请看这里](http://ksria.com/simpread/docs/#/Github)。

**Comment by Kenshin** [comment]
# 进阶教程

## 自定义

- `<repo>/src/custom.js` 中设定了显示在文章列表顶部的文章名称和 slogan。

- `<repo>/src/custom.css` 定义了主题色，可以自行更改。并可在其后附上自定义的 CSS，将会全局生效。
